packages/@aws-cdk/cdk/lib/core/construct.ts

```diff
@@ -626,9 +648,23 @@ export class Construct implements IConstruct {
    * understand the implications.
    */
   protected prepare(): void {
-    // Intentionally left blank
+    return;
   }
 
+  /**
+   * Synthesizes this construct into artifacts.
+   *
+   * This method can be overloaded by any construct that wishes to emit artifacts during
+   * the tree synthesis. For example, the `Stack` construct overrides this and produces
+   * CloudFormation templates, `Asset` overrides this to produce asset artifacts, etc.
+   *
+   * To emit artifacts, use the API of the `Session` argument.
+   *
+   * @param _session synthesis session
+   */
+  protected synthesize(_session: ISynthesisSession): void {
```

Comment:
`ISynthesizable`?

Reply:
It's more of "synthesis context" but I didn't want to overload

Reply:
No I mean, shouldn't we put the `synthesize()` method on its own interface `ISynthesizable` and search for constructs that implement that, rather than putting it on on the base `Construct` class?

Reply:
Why not, good idea

Reply:
Done

Reply:
I think @sam-goodwin is going to like this better too

Reply:
Fixed